arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1913,42 +1956,122 @@ int ClusterInfo::createViewCoordinator(std::string const& databaseName,
 ////////////////////////////////////////////////////////////////////////////////
 
 int ClusterInfo::dropViewCoordinator(
-  std::string const& databaseName, std::string const& viewID,
-  std::string& errorMsg) {
+    std::string const& databaseName,
+    std::string const& viewID,
+    std::string& errorMsg
+) {
+  // Transact to agency
+  AgencyWriteTransaction const trans{
+    // operations
+    {
+      { "Plan/Views/" + databaseName + "/" + viewID, AgencySimpleOperationType::DELETE_OP },
+      { "Plan/Version", AgencySimpleOperationType::INCREMENT_OP }
+    },
+    // preconditions
+    {
+      { "Plan/Databases/" + databaseName, AgencyPrecondition::Type::EMPTY, false },
+      { "Plan/Views/" + databaseName + "/" + viewID, AgencyPrecondition::Type::EMPTY, false }
+    }
+  };
 
   AgencyComm ac;
-  AgencyCommResult res;
+  auto const res = ac.sendTransactionWithFailover(trans);
 
-  // Transact to agency
-  AgencyOperation delPlanCollection(
-      "Plan/Views/" + databaseName + "/" + viewID,
-      AgencySimpleOperationType::DELETE_OP);
-  AgencyOperation incrementVersion("Plan/Version",
-                                   AgencySimpleOperationType::INCREMENT_OP);
-  AgencyPrecondition precondition = AgencyPrecondition(
-      "Plan/Databases/" + databaseName, AgencyPrecondition::Type::EMPTY, false);
-  AgencyPrecondition pre2 = AgencyPrecondition(
-      "Plan/Views/" + databaseName + "/" + viewID,
-      AgencyPrecondition::Type::EMPTY, false);
-  AgencyWriteTransaction trans({delPlanCollection, incrementVersion},
-      {precondition, pre2});
-  res = ac.sendTransactionWithFailover(trans);
+  // FIXME do we really have to load plan even in case of error?
```

Comment:
Error might be timeout and the operation might have happened. So I think yes.

Reply:
Should we explicitly handle timeout case?